File: refactor.js
const fs = require('fs');
const path = require('path');

const dir = 'd:\\St Thomas Tradition\\St-Thomas-Tradition\\chapters';
const files = fs.readdirSync(dir).filter(f => f.endsWith('.html') && f !== 'title.html' && f !== 'Test.html');

for (const file of files) {
  let content = fs.readFileSync(path.join(dir, file), 'utf8');
  
  // Extract number from filename (e.g. chapter-01 -> 01)
  let numMatch = file.match(/chapter-(\d+)/);
  let num = numMatch ? numMatch[1] : '';

  // Extract h1 and subtitle
  let title = '';
  let sub = '';
  
  content = content.replace(/<h1>([\\s\\S]*?)<\\/h1>/i, (m, p1) => {
    title = p1.trim();
    return '';
  });
  content = content.replace(/<p class="chapter-subtitle">([\\s\\S]*?)<\\/p>/i, (m, p1) => {
    sub = p1.trim();
    return '';
  });

  // Construct header
  if (title || sub) {
    let header = '<header class="chapter-header">\\n';
    if (num) {
      header += '  <div class="chapter-number">' + num + '</div>\\n';
    }
    if (sub) {
      header += '  <p class="chapter-subtitle">' + sub + '</p>\\n';
    }
    if (title) {
      header += '  <h1>' + title + '</h1>\\n';
    }
    header += '</header>\\n';
    
    content = content.replace(/^(<!--.*?-->\\s*)?/i, '' + header);
  }

  // Refactor blockquote
  const bqRegex = /(<header class="chapter-header">[\\s\\S]*?)(<blockquote>[\\s\\S]*?<\\/blockquote>)/i;
  content = content.replace(bqRegex, '<div class="folio-epigraph">\\n\\n</div>');

  // Add drop-cap to first paragraph of body text
  let parts = content.split('<h2>');
  if (parts.length > 1) {
    parts[1] = parts[1].replace(/<p>/, '<p class="drop-cap">');
    content = parts.join('<h2>');
  }

  fs.writeFileSync(path.join(dir, file), content, 'utf8');
}
console.log('Processed ' + files.length + ' files');

File: refactor.py
import os
import re

directory = r"d:\St Thomas Tradition\St-Thomas-Tradition\chapters"

for filename in os.listdir(directory):
    if not filename.endswith(".html") or filename in ["title.html", "Test.html"]:
        continue
        
    filepath = os.path.join(directory, filename)
    with open(filepath, 'r', encoding='utf-8') as f:
        content = f.read()
        
    # Extract chapter number
    num_match = re.search(r'chapter-(\d+)', filename)
    num = num_match.group(1) if num_match else ""
    
    # Extract h1 and subtitle
    h1_match = re.search(r'<h1>([\s\S]*?)</h1>', content, re.IGNORECASE)
    sub_match = re.search(r'<p class="chapter-subtitle">([\s\S]*?)</p>', content, re.IGNORECASE)
    
    title = h1_match.group(1).strip() if h1_match else ""
    sub = sub_match.group(1).strip() if sub_match else ""
    
    if title or sub:
        # Remove old ones
        if h1_match:
            content = content.replace(h1_match.group(0), "")
        if sub_match:
            content = content.replace(sub_match.group(0), "")
            
        # Build new header
        header = '<header class="chapter-header">\n'
        if num:
            header += f'  <div class="chapter-number">{num}</div>\n'
        if sub:
            header += f'  <p class="chapter-subtitle">{sub}</p>\n'
        if title:
            header += f'  <h1>{title}</h1>\n'
        header += '</header>\n'
        
        # Insert at top (after any leading HTML comment)
        content = re.sub(r'^(<!--.*?-->\s*)?', r'\g<1>' + header, content, count=1, flags=re.IGNORECASE)
        
    # Blockquote wrap
    bq_pattern = r'(<header class="chapter-header">[\s\S]*?)(<blockquote>[\s\S]*?</blockquote>)'
    content = re.sub(bq_pattern, r'\1<div class="folio-epigraph">\n\2\n</div>', content, count=1, flags=re.IGNORECASE)
    
    # Drop-cap logic: split by <h2>, replace first <p> in the second part
    parts = content.split('<h2>')
    if len(parts) > 1:
        parts[1] = parts[1].replace('<p>', '<p class="drop-cap">', 1)
        content = '<h2>'.join(parts)
        
    with open(filepath, 'w', encoding='utf-8') as f:
        f.write(content)

print(f"Processed HTML files.")
